File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode l1=list1,l2=list2,l3=new ListNode(-1), l3head=l3;
        while(l1!=null && l2!=null){
            if(l1.val<=l2.val){
                l3.next=l1;
                l3=l3.next;
                l1=l1.next;
            }else{
                l3.next=l2;
                l3=l3.next;
                l2=l2.next;
            }
        }
        while(l1!=null){
            l3.next=l1;
            l3=l3.next;
            l1=l1.next;
        }
        while(l2!=null){
            l3.next=l2;
            l3=l3.next;
            l2=l2.next;
        }
        return l3head.next;
    }
    // public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
    //   ListNode l1_head=list1;
    //   ListNode l2_head=list2;
    //   ListNode l3_tail=new ListNode(0);
    //   ListNode l3_head=l3_tail;
    //   while(l1_head!=null || l2_head!=null){
    //       if(l1_head!=null && l2_head!=null){
    //       if(l1_head.val>l2_head.val){
    //           l3_tail.next=new ListNode(l2_head.val);
    //           l3_tail=l3_tail.next;
    //           l2_head=l2_head.next;
    //       }else if(l1_head.val < l2_head.val){
    //           l3_tail.next=new ListNode(l1_head.val);
    //           l3_tail=l3_tail.next;
    //           l1_head=l1_head.next;
    //       }else{
    //           l3_tail.next=new ListNode(l1_head.val);
    //           l3_tail=l3_tail.next;
    //           l3_tail.next=new ListNode(l2_head.val);
    //           l3_tail=l3_tail.next;
    //           l1_head=l1_head.next;
    //           l2_head=l2_head.next;
    //       }
    //     }else if(l1_head!=null){
    //         l3_tail.next=new ListNode(l1_head.val);
    //         l3_tail=l3_tail.next;
    //         l1_head=l1_head.next;
    //     }else if(l2_head!=null){
    //         l3_tail.next=new ListNode(l2_head.val);
    //         l3_tail=l3_tail.next;
    //         l2_head=l2_head.next;
    //     }

    //   }  
    //   return l3_head.next;
    // }
}

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        list3=ListNode(-1)
        list3head=list3
        list1head=list1
        list2head=list2
        while list1head and list2head:
            nextNode=None
            if list1head.val<list2head.val:
                nextNode=list1head
                list1head=list1head.next
            else:
                nextNode=list2head
                list2head=list2head.next

            list3head.next=nextNode
            list3head=list3head.next

        while list1head:
            nextNode=list1head
            list1head=list1head.next

            list3head.next=nextNode
            list3head=list3head.next       
        
        while list2head:
            nextNode=list2head
            list2head=list2head.next
            
            list3head.next=nextNode
            list3head=list3head.next      

        return list3.next